src/app/modules/datagrid/datagrid-controller.js

```diff
@@ -20,116 +21,51 @@
 	angular.module('BloomLibraryApp.datagrid')
 	.controller('DataGridCtrl', ['$scope', '$timeout', 'bookService', '$state', '$stateParams', '$location',
 		function ($scope, $timeout, bookService, $state, $stateParams, $location) {
+			$scope.getBooks = function() {
+				var first = 0;
+				var count = bookService.getAllBooksCount();
 
-			$scope.filterOptions = {
-				filterText: "", // gets updated by user action in ngGrid
-				useExternalFilter: true
-			};
-			$scope.totalServerItems = 0;
-			$scope.pagingOptions = {
-				pageSizes: [20, 50, 100],
-				pageSize: 20, // Gets updated by user action in ngGrid
-				currentPage: 1 // Gets updated by user action in ngGrid
-			};
-			$scope.sortInfo = {
-				fields: [], directions: [], columns: []
-			};
-			$scope.getBookRange = function (count, currentPage, searchText) {
-				var first = (currentPage - 1) * count;
-				var sortField = $scope.sortInfo.fields[0];
-				var sortBy = null;
-				// Todo: setting sortBy to a complex field like this does not work...no sorting happens.
-				// It appears we will need to put a redundant top-level data field in the record for
-				// anything we want to sort by. This is especially annoying in that it may not actually
-				// be useful to sort by all the fields...but ng-grid allows the user to do so.
-				// Possibly we could disallow some fields with a message and switch to some other sort.
-				// Another reason a redundant field is probably necessary is that parse.com's built-in sorting
-				// is case-sensitive, and their only suggestion for overcoming this is a redundant
-				// all-lower-case field. Similarly any other sort-key field must somehow be in a form
-				// where a plain binary sort will give the right results.
-				if (sortField == 'title') {
-					sortBy = 'volumeInfo.title';
-				}
-				bookService.getFilteredBookRange(first, count, searchText, "", "", "", sortBy).then(function (result) {
-					$scope.visibleBooks = result;
-					$scope.visibleData = $scope.visibleBooks.map(function (item) {
+				bookService.getFilteredBookRange(first, count, '', '', '', '', '', '', true).then(function (result) {
```

Comment:
We went with grabbing them all at once because we understood that John Hatton wanted filters on every column with the new ui-grid. It was simpler to implement the filters using the API provided with ui-grid. Right now, the query is only run once, and then the ui-grid handles all of the filtering.